src/utils/assertLocationUtils.tsx

```diff
@@ -23,7 +24,13 @@ export const assertLocationTxn = async (
     return undefined
   }
 
-  const { speculativeNonce = 0 } = await getHotspotDetails(gateway)
+  let speculativeNonce = 0
+  try {
+    const response = await getHotspotDetails(gateway)
```

Comment:
I'm pretty sure this call to `getHotspotDetails` was causing location assertion to fail. It throws a 404 when onboarding a new hotspot. Wrapping it in a try/catch should fix that.

Reply:
Interesting, yeah there was a user complaining about this error. Wonder why it wasn't popping up before but this change looks good.